src/compiler/checker.ts

```diff
@@ -15071,6 +15071,11 @@ namespace ts {
             let flags = 0;
             for (const modifier of node.modifiers) {
                 switch (modifier.kind) {
+                    case SyntaxKind.ConstKeyword:
+                        if (node.kind !== SyntaxKind.EnumDeclaration) {

```

Comment:
Check if it belongs to a class, otherwise you might deliver the wrong message in other circumstances where we parse out modifiers.